Source/CarthageKit/Project.swift

```diff
@@ -557,12 +563,30 @@ public final class Project {
 			.then(.empty)
 	}
 
-	/// Attempts to build each Carthage dependency that has been checked out.
+	/// Attempts to build each Carthage dependency that has been checked out, 
+	/// optionally they are limited by the given list of dependency names.
 	///
 	/// Returns a producer-of-producers representing each scheme being built.
-	public func buildCheckedOutDependenciesWithConfiguration(configuration: String, forPlatforms platforms: Set<Platform>, sdkFilter: SDKFilterCallback = { .Success($0.0) }) -> SignalProducer<BuildSchemeProducer, CarthageError> {
-		return loadResolvedCartfile()
-			.flatMap(.Merge) { resolvedCartfile in SignalProducer(values: resolvedCartfile.dependencies) }
+	public func buildCheckedOutDependenciesWithConfiguration(configuration: String, targetDependencies: [String] = [], forPlatforms platforms: Set<Platform>, sdkFilter: SDKFilterCallback = { .Success($0.0) }) -> SignalProducer<BuildSchemeProducer, CarthageError> {
+		return buildDependencies(
+			loadResolvedCartfile()
+				.flatMap(.Merge) { resolvedCartfile -> SignalProducer<Dependency<PinnedVersion>, CarthageError> in
+					let dependencies = resolvedCartfile.dependencies
+
+					if targetDependencies.isEmpty {
+						return .init(values: dependencies)
+					} else {
+						return .init(values: dependencies.filter { targetDependencies.contains($0.project.name) })
+					}
+				},
+			withConfiguration: configuration,
+			forPlatforms: platforms,
+			sdkFilter: sdkFilter
+		)
+	}
+
+	private func buildDependencies(dependencies: SignalProducer<Dependency<PinnedVersion>, CarthageError>, withConfiguration configuration: String, forPlatforms platforms: Set<Platform>, sdkFilter: SDKFilterCallback) -> SignalProducer<BuildSchemeProducer, CarthageError> {
```

Comment:
I'd prefer to pass in a `[Dependency<PinnedVersion>]` here. The caller can easily handle `collect`ing and passing them in.